db/src/migration.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{db::VERSION_KEY, internal_error, Result};
+use ckb_logger::info;
+use rocksdb::{
+    ops::{Get, Put},
+    OptimisticTransactionDB,
+};
+
+#[derive(Default)]
+pub struct Migrations {
+    migrations: Vec<Box<dyn Migration>>,
+}
+
+impl Migrations {
+    pub fn add_migration(&mut self, migration: Box<dyn Migration>) {
+        self.migrations.push(migration);
+    }
+
+    pub fn migrate(&self, db: &OptimisticTransactionDB) -> Result<()> {
+        let db_version = db
+            .get(VERSION_KEY)
+            .map_err(|err| {
+                internal_error(format!("failed to get the version of database: {}", err))
+            })?
+            .map(|version_bytes| unsafe { String::from_utf8_unchecked(version_bytes.to_vec()) });
+
+        match db_version {
+            Some(v) => self
```

Comment:
We also need to check that if there's no applicable migrations, v must equal to the version of the last migration.